packages/ckeditor5-link/src/autolink.js

```diff
@@ -222,9 +224,12 @@ export default class AutoLink extends Plugin {
 
 		// Enqueue change to make undo step.
 		model.enqueueChange( writer => {
-			const linkHrefValue = isEmail( url ) ? `mailto:${ url }` : url;
+			const defaultProtocol = this.editor.config.get( 'link.defaultProtocol' );
+			const protocol = isEmail( url ) ? 'mailto:' : defaultProtocol;
+			const isProtocolNeeded = !!protocol && !PROTOCOL_REG_EXP.test( url );
+			const parsedValue = url && isProtocolNeeded ? protocol + url : url;
```

Comment:
I didn't notice that before - this whole logic is duplicated in \`linkui.js\`.

I think that we can extract this logic instead of \`isEmail()\`.